marlowe-runtime/web/Language/Marlowe/Runtime/Web/API.hs

```diff
@@ -84,10 +85,28 @@ instance HasNamedLink ContractHeader API "transactions" where
     "contracts" :> Capture "contractId" TxOutRef :> "transactions" :> GetTransactionsAPI
   namedLink _ _ mkLink ContractHeader{..} = guard (status == Confirmed) $> mkLink contractId
 
+data CreateTxJSON
+
+instance Accept CreateTxJSON where
+    contentType _ = "application" // "vendor.iog.marlowe-runtime.create-tx-json"
+
+instance MimeRender CreateTxJSON PostContractsCreateTxResponse where
+  mimeRender _ = encode . toJSON
+
+type PostContractsCreateTxResponse = WithLink "contract" CreateTx
+
+instance HasNamedLink CreateTx API "contract" where
+  type Endpoint CreateTx API "contract" =
+    "contracts" :> Capture "contractId" TxOutRef :> GetContractAPI
+  namedLink _ _ mkLink CreateTx{..} = Just $ mkLink contractId
+
+instance MimeUnrender CreateTxJSON PostContractsCreateTxResponse where
+  mimeUnrender _ bs = eitherDecode bs
+
```

Comment:
I wonder if instead of using a custom mime type here, the `POST` responses should just return an object with both the txBody and the tx.

```
{
  "tx": <text-envelope>,
  "txBody": <text-envelope>,
}
```

Reply:
Cool, that would simplify the flow a lot. I can switch to this simpler approach right away.